python/taichi/lang/struct.py

```diff
@@ -698,6 +705,8 @@ def filled_with_scalar(self, value):
         return Struct(entries)
 
     def field(self, **kwargs):
+        if "ndim" in kwargs:
+            kwargs.pop("ndim")
```

Comment:
Looks like `ndim` is an unexpected parameter in this case? Maybe we should throw an error instead?

Reply:
This is the case here:
https://github.com/taichi-dev/taichi/blob/9d6adb29ae7fb0b5d15234051d1f2063d8866e98/python/taichi/lang/struct.py#L328-L334

this could be any field
Maybe I could change it to checking whether it's `Struct` field, if so just ignore `ndim`.

Reply:
Yup, checking `Struct` field sounds better to me since it's a more defined behaviour. Dropping `ndim` regardlessly sounds pretty dangerous though